etna/reconciliation/base.py

```diff
@@ -83,18 +85,29 @@ def reconcile(self, ts: TSDataset) -> TSDataset:
         current_level_segments = ts.hierarchical_structure.get_level_segments(level_name=self.source_level)
         target_level_segments = ts.hierarchical_structure.get_level_segments(level_name=self.target_level)
 
+        target_names = tuple(get_target_with_quantiles(columns=ts.columns))
+        if ts.target_components_names is not None:
+            target_names += tuple(set(ts.target_components_names))
+
         df_reconciled = get_level_dataframe(
-            df=ts.to_pandas(),
+            df=ts.to_pandas(features=target_names),
             mapping_matrix=self.mapping_matrix,
             source_level_segments=current_level_segments,
             target_level_segments=target_level_segments,
         )
 
+        target_components_df = None
+        if ts.target_components_names is not None:
+            target_components_df = df_reconciled.loc[:, pd.IndexSlice[:, ts.target_components_names]]
+            df_reconciled = df_reconciled.drop(columns=ts.target_components_names, level="feature")
+
         ts_reconciled = TSDataset(
             df=df_reconciled,
             freq=ts.freq,
             df_exog=ts.df_exog,
             known_future=ts.known_future,
             hierarchical_structure=ts.hierarchical_structure,
         )
+        if ts.target_components_names is not None:
+            ts_reconciled.add_target_components(target_components_df=target_components_df)
```

Comment:
May be we should make `add_target_components` immute to `None` value.

Reply:
We can create private method which can handle `None`, but not public

Reply:
Not sure it is useful